src/lib/comms/sol-network-linux.c

```diff
@@ -236,35 +236,36 @@ _on_event(void *data, int nl_socket, unsigned int cond)
         return false;
     }
 
-    status = recvmsg(nl_socket, &msg, MSG_WAITALL);
-    if (status <= 0) {
-        if (errno == EWOULDBLOCK || errno == EAGAIN)
-            return true;
+    while ((status = recvmsg(nl_socket, &msg, MSG_WAITALL))) {
+        if (status < 0) {
+            if (errno == EWOULDBLOCK || errno == EAGAIN)
```

Comment:
`EAGAIN` and `EWOULDBLOCK` are the same under Linux (and this is a Linux-specific part of the code), so just handle one of them. While you're at it, also handle `EINTR`.